File: db/java/app/src/main/java/com/example/sales/domain/model/Payment.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 支払データのEntityクラス
 */
public class Payment {
    private String paymentNo;
    private Integer paymentDate;
    private String departmentCode;
    private LocalDateTime startDate;
    private String supplierCode;
    private Integer supplierBranch;
    private Integer paymentMethodType;
    private Integer paymentAmount;
    private Integer consumptionTax;
    private Integer completeFlag;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public Payment() {}

    // Getter/Setter（全フィールド）
    public String getPaymentNo() { return paymentNo; }
    public void setPaymentNo(String paymentNo) { this.paymentNo = paymentNo; }

    public Integer getPaymentDate() { return paymentDate; }
    public void setPaymentDate(Integer paymentDate) { this.paymentDate = paymentDate; }

    public String getDepartmentCode() { return departmentCode; }
    public void setDepartmentCode(String departmentCode) {
        this.departmentCode = departmentCode;
    }

    public LocalDateTime getStartDate() { return startDate; }
    public void setStartDate(LocalDateTime startDate) { this.startDate = startDate; }

    public String getSupplierCode() { return supplierCode; }
    public void setSupplierCode(String supplierCode) {
        this.supplierCode = supplierCode;
    }

    public Integer getSupplierBranch() { return supplierBranch; }
    public void setSupplierBranch(Integer supplierBranch) {
        this.supplierBranch = supplierBranch;
    }

    public Integer getPaymentMethodType() { return paymentMethodType; }
    public void setPaymentMethodType(Integer paymentMethodType) {
        this.paymentMethodType = paymentMethodType;
    }

    public Integer getPaymentAmount() { return paymentAmount; }
    public void setPaymentAmount(Integer paymentAmount) {
        this.paymentAmount = paymentAmount;
    }

    public Integer getConsumptionTax() { return consumptionTax; }
    public void setConsumptionTax(Integer consumptionTax) {
        this.consumptionTax = consumptionTax;
    }

    public Integer getCompleteFlag() { return completeFlag; }
    public void setCompleteFlag(Integer completeFlag) {
        this.completeFlag = completeFlag;
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/csharp/SalesManagement.Domain/Models/Payment.cs
namespace SalesManagement.Domain.Models;

/// <summary>
/// 支払データ
/// </summary>
public class Payment
{
    /// <summary>
    /// 支払伝票番号
    /// </summary>
    public string PaymentSlipNo { get; set; } = string.Empty;

    /// <summary>
    /// 支払日
    /// </summary>
    public DateTime PaymentDate { get; set; }

    /// <summary>
    /// 仕入先コード
    /// </summary>
    public string SupplierCode { get; set; } = string.Empty;

    /// <summary>
    /// 支払額
    /// </summary>
    public decimal PaymentAmount { get; set; }

    /// <summary>
    /// 支払方法 (1:現金 2:振込 3:手形 4:相殺)
    /// </summary>
    public int PaymentMethod { get; set; }

    /// <summary>
    /// 支払完了フラグ (0:未完了 1:完了)
    /// </summary>
    public int CompletedFlag { get; set; }

    /// <summary>
    /// 備考
    /// </summary>
    public string? Remarks { get; set; }

    /// <summary>
    /// 部門コード
    /// </summary>
    public string DepartmentCode { get; set; } = string.Empty;

    /// <summary>
    /// 作成日時
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 作成者名
    /// </summary>
    public string CreatedBy { get; set; } = string.Empty;

    /// <summary>
    /// 更新日時
    /// </summary>
    public DateTime UpdatedAt { get; set; }

    /// <summary>
    /// 更新者名
    /// </summary>
    public string UpdatedBy { get; set; } = string.Empty;
}
